src/stateMachine.coffee

```diff
@@ -204,38 +215,92 @@ class AutoTester extends NodeState
 				log.info '[STATE] ' + @current_state_name
 				#wait 30seconds for the power to be applied
 				@wait 30000
-				physicalMedia.powerSlave()
-				#emit event here: event: slave-powered-up
+				physicalMedia.powerSlaveWithBootMedia()
 				# TODO: need to have a GPIO input to check that power actually applied
 				@goto 'DeviceOnDashboard'
 
 			WaitTimeout: (timeout, data) ->
-				fsm = this
 				error = 'Power was never applied'
 				@goto 'ErrorState', { error: error, state: fsm.current_state_name }
 
 		DeviceOnDashboard:
-			# jenkins should wait about 2-3 minute for device to pop up in dash
 			Enter: (data) ->
 				fsm = this
 				log.info '[STATE] ' + @current_state_name
 				#start a timer, timeout after 4 minutes of waiting
-
-				poll().timeout(240000).then (uuid) ->
+				pollApp().timeout(240000).then (uuid) ->
 					log.info 'A device was found: ' + uuid
 					config.lastEvent = 'rpi booted'
-					fsm.goto 'TestSuccess'
+					# TODO: Need a more generic way of doing switch between devic-types
+					if config.img.devType == 'nuc'
+						fsm.goto 'PostProvision' , { UUID: uuid }
+					else
+						fsm.goto 'TestSuccess'
 				.catch Promise.TimeoutError, (error) ->
-					shouldPoll = false
+					shouldPollApp = false
 					fsm.goto 'ErrorState', { error: error, state: fsm.current_state_name }
 				.catch (error) ->
 					fsm.goto 'ErrorState', { error: error, state: fsm.current_state_name }
 
 			WaitTimeout: (timeout, data) ->
-				fsm = this
 				error = 'Device never showed up on dashboard'
 				@goto 'ErrorState', { error: error, state: fsm.current_state_name }
 
+		# Flasher Types Only
+		PostProvision:
+			Enter: (data) ->
+				fsm = this
+				log.info '[STATE] ' + @current_state_name
+				@wait 10 * 60 * 1000, data
+
+				pollDevice(data.UUID, provisioning_state: 'Post-Provisioning' )
+				.then (device) ->
```

Comment:
you can use fat arrows (=>) here, and in every function where you need the `fsm` var

fat arrows keep `this` variable to be the same in the inner function, so you can still use `@goto` etc